gateway/api/user/user.go

```diff
@@ -581,6 +581,177 @@ func ListAllGroups(c *gin.Context) {
 	c.JSON(http.StatusOK, groupsList)
 }
 
+// CreateGroup
+//
+//	@Summary		Create User Group
+//	@Description	Create a new user group
+//	@Tags			User Management
+//	@Accept			json
+//	@Produce		json
+//	@Param			request	body	openapi.UserGroup	true	"Group object"
+//	@Success		201		{object}	openapi.UserGroup	"{"name": "string"}"
+//	@Failure		400		{object}	openapi.HTTPError	"{"message": "error message"}"
+//	@Failure		409		{object}	openapi.HTTPError	"{"message": "group already exists"}"
+//	@Failure		500		{object}	openapi.HTTPError	"{"message": "server error"}"
+//	@Router			/users/groups [post]
+func CreateGroup(c *gin.Context) {
+	ctx := storagev2.ParseContext(c)
+
+	var req openapi.UserGroup
+
+	if err := c.ShouldBindJSON(&req); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message": err.Error()})
+		return
+	}
+
+	// Check if group already exists
+	userGroups, err := models.GetUserGroupsByOrgID(ctx.OrgID)
+	if err != nil {
+		log.Errorf("failed getting org groups, err=%v", err)
+		sentry.CaptureException(err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "failed getting org groups"})
+		return
+	}
+
+	for _, ug := range userGroups {
+		if ug.Name == req.Name {
+			c.JSON(http.StatusConflict, gin.H{"message": fmt.Sprintf("group %s already exists", req.Name)})
+			return
+		}
+	}
+
+	if err := models.CreateUserGroupWithoutUser(ctx.OrgID, req.Name); err != nil {
+		log.Errorf("failed creating group, err=%v", err)
+		sentry.CaptureException(err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "failed creating group"})
+		return
+	}
+
+	c.JSON(http.StatusCreated, req)
+}
+
+// UpdateGroup
+//
+//	@Summary		Update User Group
+//	@Description	Update an existing user group name
+//	@Tags			User Management
+//	@Accept			json
+//	@Produce		json
+//	@Param			name	path	string	true	"The name of the group to update"
+//	@Param			request	body	openapi.UserGroup	true	"Group object with new name"
+//	@Success		200		{object}	openapi.UserGroup	"{"name": "string"}"
+//	@Failure		400		{object}	openapi.HTTPError	"{"message": "error message"}"
+//	@Failure		404		{object}	openapi.HTTPError	"{"message": "group not found"}"
+//	@Failure		409		{object}	openapi.HTTPError	"{"message": "group already exists"}"
+//	@Failure		500		{object}	openapi.HTTPError	"{"message": "server error"}"
+//	@Router			/users/groups/{name} [put]
+func UpdateGroup(c *gin.Context) {
```

Comment:
This route will perform a delete and insert operation which doesn't comply with the REST method PUT.
The `name` attribute has a unique constraint meaning it's used as a unique identifier and you're permitting to change this unique constraint. Therefore, in practice you're going to remove the record and add a new one.

I think it will be better to remove this route and allow clients to perform a DELETE and POST operation in this case which is more explicit.